pytorch_lightning/trainer/trainer.py

```diff
@@ -855,11 +924,18 @@ def tune(
 
     def call_setup_hook(self, model):
         # call setup after the ddp process has connected
-        stage_name = 'test' if self.testing else 'fit'
+        stage_name = self.evaluating or 'fit'
+
         if self.datamodule is not None:
-            called = self.datamodule.has_setup_test if self.testing else self.datamodule.has_setup_fit
+            called = {
+                None: self.datamodule.has_setup_fit,
+                'validation': self.datamodule.has_setup_validation,
+                'test': self.datamodule.has_setup_test,
+            }[self.evaluating]
```

Comment:
not a big fan of `{None: 'something`}. Mind put this in simple if-else conditionals.